### PR TITLE
Update Java and C++ style guides with various changes since last update

### PR DESCRIPTION
**Pull request by shicks**

### CODE REVIEW

**Comment by shicks** [comment]
Linking to the internal style groups to make sure there's buy-in before we merge this.

File: cppguide.html
Comment (mzeren-vmw) [inline cppguide.html]: I think we want the LHS github URL.

File: cppguide.html
Comment (mzeren-vmw) [inline cppguide.html]: Do we really want to drop https? Double check other URLs in this edit. Also this particular URL should be https://github.com/google/googletest.

File: cppguide.html
Comment (shicks) [reply cppguide.html]: This has been reverted. The internal version of this file had gotten out of sync.

File: cppguide.html
Comment (shicks) [reply cppguide.html]: Same here - this should be fixed now.

Comment (mzeren-vmw) [reply cppguide.html]: Looks good.

**Comment by geoffromer** [comment]
LGTM (C++ only, didn't look at Java)

**Comment by tkruse** [comment]
BTW, without wanting to spoil all the fun of manually editing HTML, maybe any of the google maintainers would be interested in converting all guidelines to markdown? As I sketched here a long time ago: #36

**Comment by shicks** [comment]
Internally the new JS guide _is_ markdown. Unfortunately, or internal
markdown renderer is a flavor of hoedown, whereas github pages uses
kramdown, so we end up converting it to HTML on the way out anyway...

**Comment by tkruse** [comment]
Interesting. But still how is html better than hoedown on github?

**Comment by shicks** [comment]
Because we can actually produce it and have it render correctly. Our style
guides take significant advantage of one of the nonstandard markdown
features (classes on code blocks and headings) which is unfortunately
incompatible between the two engines. So we'd need to do some sort of
postprocessing on it in any case, which is more moving parts that don't
exist yet and would need maintenance. Additionally, our externalization
process only works on HTML, so we'd need to devise a way to scrub markdown
directly, which also doesn't exist yet.

It's a great feature request to work toward, and I'd personally love to see
it happen, but given that our style guides are all volunteer/20% efforts,
it's not likely to happen in the near term.

On Nov 5, 2016 9:58 PM, "Thibault Kruse" notifications@github.com wrote:

Interesting. But still how is html better than hoedown on github?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
https://github.com/google/styleguide/pull/196#issuecomment-258660720, or mute
the thread

**Comment by tkruse** [comment]
So this "externalization process" cut's away pieces that must not be open-source for now? 

Or could Google publish the raw hoedown and externalization scripts on github as well as the HTML so that contributors may both contribute to the raw sources directly as well as work on the transformation?

Additionally, what about the guides provided as xml files (objc, shell, lisp, ...), are they internally in hoedown, or is it interesting for them to be converted to markdown?

**Review by Foxytine31 — APPROVED** [review]

**Comment by shicks** [comment]
Internal Java people are okay with this.  Can somebody merge it?

**Pull request merged**
